victim-loss-identifier/src/agent.ts

```diff
@@ -87,15 +98,56 @@ export function provideHandleAlert(): HandleAlert {
               alertEvent.blockNumber!
             ))
           );
-          break;
-        
-        case "SCAM-DETECTOR-ICE-PHISHING":
-          findings.push();
-          break;
-
-        default:
-          return findings;
-      }
+        }
+        break;
+
+      case "SCAM-DETECTOR-ICE-PHISHING":
+        const { botId: sourceAlertBotId, hash: sourceAlertHash } = alertEvent.alert.source!.sourceAlert!;
+        const underlyingAlert = (await getAlerts({ alertHash: sourceAlertHash })).alerts[0];
+
+        // E.g. npm run alert 0xd5222709dfb9a6291296cf71bf5a2aec661e8950ef1ea257edf3e9d3af8899ec
+        if (!underlyingAlert) break;
+
+        const { alertId: underlyingAlertID, metadata, description } = underlyingAlert;
+
+        if (!ICE_PHISHING_ALERT_IDS.includes(underlyingAlertID!)) break;
+
+        const scammerAddresses: string[] = [];
+
+        if (sourceAlertBotId === NETHERMIND_ICE_PHISHING_BOT) {
+          if (underlyingAlertID !== "ICE-PHISHING-HIGH-NUM-APPROVED-TRANSFERS") {
+            scammerAddresses.push(scammerAddress);
+          } else {
+            const underlyingAlertFirstTxHash = metadata["firstTxHash"];
+            const underlyingAlertLastTxHash = metadata["lastTxHash"];
+            const [firstTxReceipt, lastTxReceipt] = await Promise.all([
+              dataFetcher.getTransactionReceipt(underlyingAlertFirstTxHash),
+              dataFetcher.getTransactionReceipt(underlyingAlertLastTxHash),
+            ]);
+
+            extractScammerAddresses(firstTxReceipt!, scammerAddresses);
+            extractScammerAddresses(lastTxReceipt!, scammerAddresses);
+          }
+        } else if (sourceAlertBotId === BLOCKSEC_ICE_PHISHING_BOT && !description!.includes("approve")) {
+          // Checking if scammer address exists as there exist alerts about phishing urls and not EOAs, where the property is empty.
```

Comment:
The comment at L132 does what you say, the first check is about excluding the approval alerts and keep only the transfer ones.

Reply:
Is there a reason we wouldn't want to include scammers from the `Ice Phishing WEB Alert`?

Reply:
@RCantu92, I think those alerts are e.g. like this https://explorer.forta.network/alert/0x34f8a1155cbbe230b474228096742f4c1c7d959a5eeed0ebe04273c65002e6ea (although the source alert is "omitted" and not "Ice-phishing-web")

Have you seen any `Ice-phishing-web` spefically? Even `getAlerts` doesn't return alerts (although I may be using a wrong query there, e.g. wrong block number range)

In any case, we aren't really able to process URLs as we don't have an address (except if I'm missing something?)

Reply:
I have not, but I see what you mean by it being omitted. And I really only meant that we could monitor the addresses deemed a `scammer` in those alerts to see if they were recipients of Transfers.

However, similar to the _Spam Token_ bot, it may not have no connection to the activity after a victim gets phished via a URL. But mostly I was thinking we could still monitor in case there were transfers to them.

Reply:
@RCantu92, note that at least on the alerts that I'm seeing which contain a URL as the scammer, the address property is empty

Reply:
I see it now. Thanks for clarifying. 👍